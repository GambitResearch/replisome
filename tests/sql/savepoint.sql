\set VERBOSITY terse
\pset format unaligned

-- predictability
SET synchronous_commit = on;

CREATE TABLE xpto (a SERIAL PRIMARY KEY, b text);

SELECT slot_create();

INSERT INTO xpto (b) VALUES('john');
INSERT INTO xpto (b) VALUES('smith');
INSERT INTO xpto (b) VALUES('robert');

BEGIN;
INSERT INTO xpto (b) VALUES('marie');
SAVEPOINT sp1;
INSERT INTO xpto (b) VALUES('ernesto');
SAVEPOINT sp2;
INSERT INTO xpto (b) VALUES('peter');	-- discard
SAVEPOINT sp3;
INSERT INTO xpto (b) VALUES('albert');	-- discard
ROLLBACK TO SAVEPOINT sp2;
RELEASE SAVEPOINT sp1;
INSERT INTO xpto (b) VALUES('francisco');
END;

SELECT data FROM slot_get();
SELECT slot_drop();
