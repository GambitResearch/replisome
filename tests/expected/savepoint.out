\set VERBOSITY terse
\pset format unaligned
-- predictability
SET synchronous_commit = on;
CREATE TABLE xpto (a SERIAL PRIMARY KEY, b text);
SELECT slot_create();
slot_create
init
(1 row)
INSERT INTO xpto (b) VALUES('john');
INSERT INTO xpto (b) VALUES('smith');
INSERT INTO xpto (b) VALUES('robert');
BEGIN;
INSERT INTO xpto (b) VALUES('marie');
SAVEPOINT sp1;
INSERT INTO xpto (b) VALUES('ernesto');
SAVEPOINT sp2;
INSERT INTO xpto (b) VALUES('peter');	-- discard
SAVEPOINT sp3;
INSERT INTO xpto (b) VALUES('albert');	-- discard
ROLLBACK TO SAVEPOINT sp2;
RELEASE SAVEPOINT sp1;
INSERT INTO xpto (b) VALUES('francisco');
END;
SELECT data FROM slot_get();
data
{
	"tx": [
		{
			"op": "I",
			"schema": "public",
			"table": "xpto",
			"colnames": ["a", "b"],
			"coltypes": ["int4", "text"],
			"values": [1, "john"]
		}
	]
}
{
	"tx": [
		{
			"op": "I",
			"schema": "public",
			"table": "xpto",
			"values": [2, "smith"]
		}
	]
}
{
	"tx": [
		{
			"op": "I",
			"schema": "public",
			"table": "xpto",
			"values": [3, "robert"]
		}
	]
}
{
	"tx": [
		{
			"op": "I",
			"schema": "public",
			"table": "xpto",
			"values": [4, "marie"]
		}
		,{
			"op": "I",
			"schema": "public",
			"table": "xpto",
			"values": [5, "ernesto"]
		}
		,{
			"op": "I",
			"schema": "public",
			"table": "xpto",
			"values": [8, "francisco"]
		}
	]
}
(4 rows)
SELECT slot_drop();
slot_drop
stop
(1 row)
